File: HW2/Tests/ProgramTester.cs
using Xunit;
using static HW2.Program;

namespace Tests
{
    public class ProgramTester
    {
        [Theory]
        [InlineData(false, new [] {"1", "5"})]
        [InlineData(false, new [] {"+", "10", "bug"})]
        [InlineData(false, new [] {"3", "/", "0"})]
        public void IsNotValidArgs(bool isValid, string[] args)
        {
            Assert.False(Main(args) == 0);
        }

        [Theory]
        [InlineData(true, new [] {"1", "+", "10"})]
        [InlineData(true, new [] {"3", "/", "1"})]
        [InlineData(true, new [] {"1", "*", "20"})]
        public void ValidArgs(bool isValid, string[] args)
        {
            Assert.True(Main(args) == 0);
        }
    }
}

File: HW4/Tests/ProgramTester.fs
module Tests.ProgramTester

open Xunit
open HW4

let validArguments = [|
    [|[|"14"; "+"; "3"|]|]
    [|[|"19"; "-"; "101"|]|]
    [|[|"47"; "*"; "17"|]|]
    [|[|"24"; "/"; "489"|]|]
    [|[|"1134"; ":"; "98541"|]|]  
|]

[<Theory>]
[<MemberData(nameof(validArguments))>]
let ``MainWithValidArgs`` args =
    Assert.True(Program.main(args) = 0)
    
let invalidArguments = [|
    [|[|"1"; "+"; "2"; "="|]|]
    [|[|"131.0"; "-"; "2"|]|]
    [|[|"12"; "*"; "hello"|]|]
    [|[|"144"; "/"; "0"|]|]
    [|[|"131"; ":"; "0"|]|]
|]

[<Theory>]
[<MemberData(nameof(invalidArguments))>]
let ``MainWithInvalidArgs`` args =
    Assert.False(Program.main(args) = 0)
